JMPriceSuggestion.lua

```diff
@@ -199,77 +199,81 @@ local AlgorithmFunctionList = {
     -- Marcus' normal
     --
     [Algorithms.NORMAL] = function(guildSaleList, allSaleList)
-        local function probit(x)
-            local function inverseError(x)
-                local a = 0.147
-
-                if x == 0 then
-                    return 0
+        local function getPrice(quantile, saleList) 
+            local function probit(x)
+                local function inverseError(x)
+                    local a = 0.147
+
+                    if x == 0 then
+                        return 0
+                    end
+
+                    local log = math.log(1 - x * x)
+                    local log_div_2 = log / 2
+                    local b = log_div_2 + 2 / math.pi / a
+                    local first_root = math.sqrt(b * b - log / 2)
+                    local second_root = math.sqrt(first_root - b)
+
+                    if x > 0 then
+                        return second_root
+                    end
+
+                    return -1 * second_root
                 end
 
-                local log = math.log(1 - x * x)
-                local log_div_2 = log / 2
-                local b = log_div_2 + 2 / math.pi / a
-                local first_root = math.sqrt(b * b - log / 2)
-                local second_root = math.sqrt(first_root - b)
-
-                if x > 0 then
-                    return second_root
+                return (inverseError(x * 2 - 1) * math.sqrt(2))
+            end
+            
+            local function getAgeWeight(sale)
+                local function logWeight(hours)
+                    local log = math.log(30 + hours)
+                
+                    return math.max(1, (40 / log / log))
                 end
-
-                return -1 * second_root
+            
+                local ageSeconds = GetTimeStamp() - sale.saleTimestamp
+                local ageHours = ageSeconds / 3600
+                
+                return logWeight(ageHours)
             end
-
-            return (inverseError(x * 2 - 1) * math.sqrt(2))
-        end
-        
-        local function getAgeWeight(sale)
-            local function logWeight(hours)
-                local log = math.log(20 + hours)
             
-                return math.max(1, (40 / log / log))
+            if #saleList < 2 then
+                return math.ceil(allSaleList[1].pricePerPiece * 0.9)
```

Comment:
That can be because here you return no second value. What needs the weight be if there is only 1 sale?